### PR TITLE
refactor!: remove counterparty connection interface

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

:smile: 

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ There are four sections based on the four potential user groups of this document
 
 ### API removals
 
-The `exported.ChannelI` and `exported.CounterpartyChannelI` interfaces has been removed. Please use the concrete types. 
-The `exported.ConnectionI` interface has been removed. Please use the concrete types. 
+The `exported.ChannelI` and `exported.CounterpartyChannelI` interfaces has been removed. Please use the concrete types.
+The `exported.ConnectionI` and `exported.CounterpartyConnectionI` interfaces has been removed. Please use the concrete types.
```
Comment (colin-axner) [inline docs/docs/05-migrations/13-v8-to-v9.md:30]:
```suggestion
The `exported.ChannelI` and `exported.CounterpartyChannelI` interfaces have been removed. Please use the concrete types.
The `exported.ConnectionI` and `exported.CounterpartyConnectionI` interfaces have been removed. Please use the concrete types.
```

**Review by crodriguezvega — APPROVED** [review]
Instead of innovation week, we should call it cleaning week 😄

**Review by damiannolan — APPROVED** [review]
🗡️

**Comment by colin-axner** [comment]
> Instead of innovation week, we should call it cleaning week 😄

It'll all make sense in the end :smile:

**Pull request merged**